internal/repository/ent/connection.go

```diff
@@ -0,0 +1,531 @@
+		"tenant_id", c.TenantID,
+		"name", c.Name,
+	)
+
+	// Start a span for this repository operation
+	span := StartRepositorySpan(ctx, "connection", "create", map[string]interface{}{
+		"connection_id": c.ID,
+		"name":          c.Name,
+	})
+	defer FinishSpan(span)
+
+	// Set environment ID from context if not already set
+	if c.EnvironmentID == "" {
+		c.EnvironmentID = types.GetEnvironmentID(ctx)
+	}
+
+	// Convert structured metadata to map format for database storage
+	metadataMap := convertConnectionMetadataToMap(c.Metadata, c.ProviderType)
+
+	connection, err := client.Connection.Create().
+		SetID(c.ID).
+		SetTenantID(c.TenantID).
+		SetName(c.Name).
+		SetProviderType(connection.ProviderType(c.ProviderType)).
+		SetMetadata(metadataMap).
+		SetStatus(string(c.Status)).
+		SetCreatedAt(c.CreatedAt).
+		SetUpdatedAt(c.UpdatedAt).
+		SetCreatedBy(c.CreatedBy).
+		SetUpdatedBy(c.UpdatedBy).
+		SetEnvironmentID(c.EnvironmentID).
+		Save(ctx)
+
+	if err != nil {
+		SetSpanError(span, err)
+
+		if ent.IsConstraintError(err) {
+			return ierr.WithError(err).
+				WithHint("Failed to create connection").
+				WithReportableDetails(map[string]any{
+					"name":          c.Name,
+					"provider_type": c.ProviderType,
+				}).
+				Mark(ierr.ErrAlreadyExists)
+		}
+		return ierr.WithError(err).
+			WithHint("Failed to create connection").
+			Mark(ierr.ErrDatabase)
+	}
+
+	SetSpanSuccess(span)
+	*c = *domainConnection.FromEnt(connection)
+	return nil
+}
+
+func (r *connectionRepository) Get(ctx context.Context, id string) (*domainConnection.Connection, error) {
+	// Start a span for this repository operation
+	span := StartRepositorySpan(ctx, "connection", "get", map[string]interface{}{
+		"connection_id": id,
+	})
+	defer FinishSpan(span)
+
+	// Try to get from cache first
+	if cachedConnection := r.GetCache(ctx, id); cachedConnection != nil {
+		return cachedConnection, nil
+	}
+
+	client := r.client.Querier(ctx)
+	r.log.Debugw("getting connection", "connection_id", id)
+
+	c, err := client.Connection.Query().
+		Where(
+			connection.ID(id),
+			connection.TenantID(types.GetTenantID(ctx)),
+			connection.EnvironmentID(types.GetEnvironmentID(ctx)),
+			connection.Status(string(types.StatusPublished)),
+		).
+		Only(ctx)
+
+	if err != nil {
+		SetSpanError(span, err)
+
+		if ent.IsNotFound(err) {
+			return nil, ierr.WithError(err).
+				WithHintf("Connection with ID %s was not found", id).
+				WithReportableDetails(map[string]any{
+					"connection_id": id,
+				}).
+				Mark(ierr.ErrNotFound)
+		}
+		return nil, ierr.WithError(err).
+			WithHint("Failed to get connection").
+			Mark(ierr.ErrDatabase)
+	}
+
+	SetSpanSuccess(span)
+	domainConn := domainConnection.FromEnt(c)
+
+	// Cache the result
+	r.SetCache(ctx, domainConn)
+
+	return domainConn, nil
+}
+
+func (r *connectionRepository) GetByEnvironmentAndProvider(ctx context.Context, environmentID string, provider types.SecretProvider) (*domainConnection.Connection, error) {
```

Comment:
this should not have env id or tenant_id as input parameter and not be passed in

Reply:
Make this get by provider simpple